agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1759,6 +1764,83 @@ bool backhaul_manager::handle_1905_1_message(ieee1905_1::CmduMessageRx &cmdu_rx,
     }
 }
 
+bool backhaul_manager::handle_ap_capability_query(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                                  const std::string &src_mac)
+{
+    //TODO - this is a stub handler for the purpose of controller certification testing,
+    //       will be implemented later on agent certification
+    const auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received AP_CAPABILITY_QUERY_MESSAGE, mid=" << std::dec << int(mid);
+
+    if (!cmdu_tx.create(mid, ieee1905_1::eMessageType::AP_CAPABILITY_REPORT_MESSAGE)) {
+        LOG(ERROR) << "cmdu creation of type AP_CAPABILITY_REPORT_MESSAGE, has failed";
+        return false;
+    }
+
+    auto ap_capability_tlv = cmdu_tx.addClass<wfa_map::tlvApCapability>();
+    if (!ap_capability_tlv) {
+        LOG(ERROR) << "addClass wfa_map::tlvApCapability has failed";
+        return false;
+    }
+
+    for (const auto &radio_cap : m_radio_capabilities_map) {
+        std::vector<uint8_t> operating_classes;
+        auto ruid = radio_cap.first;
+
+        auto radio_basic_caps = cmdu_tx.addClass<wfa_map::tlvApRadioBasicCapabilities>();
```

Comment:
This is duplicated code - please use `add_radio_basic_capabilities` and refactor it accordingly if needed so you can use it from here.

Reply:
we also can pass those private members as arguments to this function.

Reply:
I've created a new class which for the moment, contains just one static method: add_ap_radio_basic_capabilities. Required private members are passed as parameters.
This class is intended to contain code to build up TLVs and to be shared between backhaul_manager_thread and son_slave_thread (or any other agent thread)